### ISSUE TITLE
Unable to launch dask-gateway cluster with distributed 2.20

### ISSUE THREAD

**Issue by scottyhq** [pangeo-data/pangeo-cloud-federation#655]
cross posting from https://github.com/pangeo-data/pangeo-docker-images/issues/96#issuecomment-657741692

i'm seeing errors in logs like this on the AWS hub `kubectl logs -n icesat2-prod dask-gateway-5f8088e361a24ad481cedca5966a2949`

distributed.core - ERROR - addresses should be strings or tuples, got None
Traceback (most recent call last):
  File "/srv/conda/envs/notebook/lib/python3.7/site-packages/distributed/core.py", line 513, in handle_comm
    result = await result
  File "/srv/conda/envs/notebook/lib/python3.7/site-packages/distributed/scheduler.py", line 2208, in remove_worker
    address = self.coerce_address(address)
  File "/srv/conda/envs/notebook/lib/python3.7/site-packages/distributed/scheduler.py", line 4946, in coerce_address
    raise TypeError("addresses should be strings or tuples, got %r" % (addr,))
TypeError: addresses should be strings or tuples, got None

@TomAugspurger ideas?
maybe related to recent dask version bumps from 2.19 to 2.20
https://github.com/pangeo-data/pangeo-docker-images/compare/2020.07.03...2020.07.12

**Comment by scottyhq** [comment]
@TomAugspurger 's comment:
> @scottyhq unsure, but the error message is the same as dask/dask#6372. Do you know if the workers are starting successfully, or are the pods terminating? In the linked issue, the worker processes didn't have write permissions to its working directory.

worker pods are terminated right when they launch, so not getting insightful logs. Not sure what directory is used for worker scratch space by default? is this something we should explicitly set here? https://github.com/pangeo-data/pangeo-cloud-federation/blob/staging/pangeo-deploy/values.yaml#L68

**Comment by scottyhq** [comment]
@TomAugspurger Linking to a PR of yours because it seems related https://github.com/dask/dask-gateway/pull/283#issuecomment-654852231. Definitely related to distributed 2.20 (running on the GCP cluster is fine because the image has 2.19).  Here is a binder repo setup I'm using to try to diagnose and fix: https://github.com/scottyhq/pangeo-notebook-staging

**Comment by scottyhq** [comment]
running `client.scheduler_info()` with distributed 2.19 shows `home/jovyan` as the local_directory so I'm not sure why there could be write permission issues:
```
 'workers': {'tls://192.168.150.39:44571': {'type': 'Worker',
   'id': 'dask-gateway-g62p6',
   'host': '192.168.150.39',
   'resources': {},
   'local_directory': '/home/jovyan/worker-8vl7lcz8',
```

**Comment by scottyhq** [comment]
a bit more diagnosis with the following quick check:
```python
def listdir():
    import subprocess
    return subprocess.run(["ls", "-l", "/home/"], capture_output=True)

result = client.run(listdir)
print(result[address].stdout.decode("utf-8"))

def getid():
    import subprocess
    return subprocess.run(["id"], capture_output=True)

result = client.run(getid)
print(result[address].stdout.decode("utf-8"))

def lsps():
    import subprocess
    return subprocess.run(["ps", "-aux"], capture_output=True)

result = client.run(lsps)
print(result[address].stdout.decode("utf-8"))


def lspid():
    import subprocess
    return subprocess.run(["pwdx", "1"], capture_output=True)

result = client.run(lspid)
print(result[address].stdout.decode("utf-8"))
```

```
total 0
drwxr-xr-x 1 jovyan jovyan 112 Jul 14 22:55 jovyan

uid=1000(jovyan) gid=1000(jovyan) groups=1000(jovyan)

USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
jovyan       1  1.3  0.1 606444 67580 ?        Ssl  22:55   0:04 /srv/conda/envs/notebook/bin/python /srv/conda/envs/notebook/bin/dask-gateway-worker --nthreads 2 --memory-limit 4294967296 --scheduler-address tls://dask-gateway-2d009dfbc1f24a54b7733c5200813aa3.prod:8786
jovyan      21  0.0  0.0  27404 11288 ?        S    22:55   0:00 /srv/conda/envs/notebook/bin/python -c from multiprocessing.semaphore_tracker import main;main(10)
jovyan      23  1.6  0.1 913504 86864 ?        Sl   22:55   0:05 /srv/conda/envs/notebook/bin/python -c from multiprocessing.spawn import spawn_main; spawn_main(tracker_fd=11, pipe_handle=17) --multiprocessing-fork
jovyan      45  0.0  0.0  36156  3284 ?        R    23:00   0:00 ps -aux

1: /home/jovyan
```

**Comment by TomAugspurger** [comment]
This should be released in distributed 2.21 on Friday.
